### PR TITLE
UHM-4867

### PR DESCRIPTION
**Pull request by mseaton**

### CODE REVIEW

**Review by mogoodrich — APPROVED** [review]
LGTM! I didn't go into the weeds, but that will come out in the testing!

**Comment by mseaton** [comment]
@brandones / @cioan I'll look to merge this in later today, but wanted to give you a chance to have a look first in case you are interested / have feedback.

**Review by cioan — APPROVED** [review]
LGTM! Thanks!

File: etl/src/main/java/org/pih/hivmigration/etl/sql/SqlMigrator.java [hunk 1]
```diff
@@ -317,4 +321,68 @@ private String abbreviate(String str) {
         }
     }
 
+    // Methods that allow for creating tests that should pass
+
+    protected void assertAllRows(String description, String mysqlQuery) throws Exception {
+        log.info("Testing: " + description);
+        Number allRows = (Number) selectMysql("select count(*) from drug_order", new ScalarHandler<Number>());
```
Comment (brandones) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/SqlMigrator.java:328]: Looks like this function is specific to the `drug_order` table. Can you generalize it, maybe with a `tableName` parameter?

File: etl/src/main/java/org/pih/hivmigration/etl/sql/SqlMigrator.java [hunk 2]
```diff
@@ -317,4 +321,68 @@ private String abbreviate(String str) {
...
+            for (String key : oracleRow.keySet()) {
+                allKeys.add(key.toLowerCase());
+            }
+            for (String key : mysqlRow.keySet()) {
+                allKeys.add(key.toLowerCase());
+            }
+            for (String key : allKeys) {
+                Object oracleValue = oracleRow.get(key.toUpperCase());
+                Object mysqlValue = mysqlRow.get(key);
+                if (oracleValue != null || mysqlValue != null) {
```
Comment (brandones) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/SqlMigrator.java:368]: Can remove this outer `if` block and just make that last `else` into an `else if(mysqlValue == null && oracleValue == null)`.

**Review by brandones — APPROVED** [review]
Awesome work. One bug and one suggestion.

**Pull request merged**